### ISSUE TITLE
Boolean format of configure_file() changed in Meson 1.3.0

### ISSUE THREAD

**Issue by ssssam** [mesonbuild/meson#12591]
**Describe the bug**
The configure_file() function changed how booleans are output between 1.2.0 and 1.3.0. This breaks existing code which expects a certain format

**To Reproduce**

Here's a minimal reproducer:

meson.build:

```
project('set-boolean-test')

cdata = configuration_data()

cdata.set('enable_foo', true)
configure_file(input: 'cdata.conf.in', output: 'cdata.conf', configuration: cdata)
```

cdata.conf.in:

```
enable_foo = @enable_foo@
```

**Expected behavior**

With Meson 1.2.0 the `cdata.conf` file in the build tree looks like this:

```
enable_foo = True
```

With Meson 1.3.0 the `cdata.conf` file looks like this:

```
enable_foo = 1
```

I don't see anything in the release notes about this change so I guess it is unintentional. And it's not great for developers for this to change in a stable series as now we have to support two different formats, or pin Meson version to <1.3.0 or >=1.3.0.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? No
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.) Fedora 38 (1.2.0) and Fedora 39 (1.3.0)
* what Python version are you using: Python 3.12 (Meson 1.3.0) and Python 3.11 (Meson 1.2.0)

**Comment by eli-schwartz** [comment]
1.3.0 is broken as heck and will be fixed in 1.3.1, the fix is merged in git master via #12532.

**Comment by ssssam** [comment]
Ok, thanks for the info :)

**Comment by amitdo** [comment]
> 1.3.0 is broken as heck and will be fixed in 1.3.1, the fix is merged in git master via #12532.

@eli-schwartz,

1.3.1 was released 5 months  ago, so why is this closed issue still pinned?

**Comment by amitdo** [comment]
ping @jpakkane
